modules/effects/src/main/scala/dev/profunktor/redis4cats/redis.scala

```diff
@@ -370,8 +369,42 @@ private[redis4cats] class BaseRedis[F[_]: Concurrent: ContextShift, K, V](
       .futureLift
       .map(x => x == key.size.toLong)
 
+  /**
+    * Expires a key with the given duration. If specified either in MILLISECONDS, MICROSECONDS or NANOSECONDS,
+    * the value will be converted to MILLISECONDS. Otherwise, it will be converted to SECONDS.
+    *
+    * As expected by Redis' PEXPIRE and EXPIRE commands, respectively.
+    */
   def expire(key: K, expiresIn: FiniteDuration): F[Unit] =
-    async.flatMap(c => F.delay(c.expire(key, expiresIn.toSeconds))).futureLift.void
+    async
+      .flatMap { c =>
+        expiresIn.unit match {
+          case TimeUnit.MILLISECONDS | TimeUnit.MICROSECONDS | TimeUnit.NANOSECONDS =>
+            F.delay(c.pexpire(key, expiresIn.toMillis))
+          case _ =>
+            F.delay(c.expire(key, expiresIn.toSeconds))
+        }
+      }
+      .futureLift
+      .void
+
+  /**
+    * Expires a key at the given date.
+    *
+    * It maps Redis' EXPIREAT and PEXPIREAT commands, depending on the given [[TimePrecision]].
+    */
+  def expireAt(key: K, at: Instant, precision: TimePrecision): F[Unit] =
```

Comment:
Not sure why you introduce the concept of "Precision" here; the user of the API has full flexibility to get to that granularity through the precision used in the `Instant` (?)

Given that expireat or pexpireat have the same cost, why not keep it simple and have it always use pexpireat?

Reply:
That's a good point. I did it to reflect the precision difference between `EXPIREAT` and `PEXPIREAT` but it seems legacy design from Redis' side. I'll simplify it to only use `PEXPIREAT` :+1: